README.md
# Financial-Calculator-JavaFx

Financial Calculator/History.txt
str2
Future Value=Rs.13128.62
Interest Rate=3
Time Period=3
Principal Value=12000
------------------------------
str4
FutureValue=20000
Principal Investment=12000
Interest Rate=2
Time Period=2
MonthlyPay=
------------------------------
str4
FutureValue=
Principal Investment=12000
Interest Rate=2
Time Period=2
MonthlyPay=326
------------------------------
str3
Down Payment=1000
Loan Term=2
Interest Rate=2
Monthly Payment=
Loan Amount=12000
------------------------------
str4
FutureValue=
Principal Investment=
Interest Rate=
Time Period=
MonthlyPay=
------------------------------
str2
Future Value=
Interest Rate=
Time Period=
Principal Value=
------------------------------
str3
Down Payment=5000
Loan Term=2
Interest Rate=7
Monthly Payment=
Loan Amount=12000
------------------------------
str2
Future Value=48997
Interest Rate=8.99 %
Time Period=10
Principal Value=20000
------------------------------


Financial Calculator/src/sample/MainPage.java
package sample;

import com.sun.java.swing.plaf.windows.WindowsInternalFrameTitlePane;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.control.PasswordField;
import javafx.scene.control.TextField;
import javafx.scene.image.ImageView;
import javafx.scene.layout.*;
import javafx.scene.text.Font;
import javafx.scene.text.FontWeight;
import javafx.scene.text.Text;
import javafx.stage.Stage;
import javafx.scene.image.Image;
import javafx.stage.Window;
import javafx.scene.paint.*;

public class MainPage extends Application {
    Stage window1;


    public static void main(String[] args) {

        launch(args);
    }

    @Override
    public void start(Stage primaryStage) throws Exception {

        window1 = primaryStage;
        window1.setTitle("Manager Log in");

        GridPane gridMainPage=new GridPane();
        gridMainPage.setAlignment(Pos.TOP_CENTER);
        gridMainPage.setVgap(10);

        gridMainPage.setPadding(new Insets(0,1,2,1));
        Image backgroundImage = new Image(getClass().getResourceAsStream("back.jpg"));

        BackgroundImage backgroundimage = new BackgroundImage(backgroundImage,
                BackgroundRepeat.NO_REPEAT,
                BackgroundRepeat.NO_REPEAT,
                BackgroundPosition.CENTER,
                BackgroundSize.DEFAULT);

        //       create Background
        Background background = new Background(backgroundimage);
        gridMainPage.setBackground(background);

        Scene scene=new Scene(gridMainPage,1050,550);
        window1.setScene(scene);
        window1.show();

        Label lblMainPageTitle=new Label("Financial Calculator");
        lblMainPageTitle.setFont(Font.font("Verdana", FontWeight.BOLD,36));
        gridMainPage.add(lblMainPageTitle,1,3);

        Button btnComSavings= new Button("Compound Savings");
        btnComSavings.setFont(Font.font("Arial", FontWeight.BOLD,24));
        gridMainPage.add(btnComSavings,1,5);


        Button btnSimSavings= new Button("Simple Savings");
        btnSimSavings.setFont(Font.font("Arial", FontWeight.BOLD,24));
        gridMainPage.add(btnSimSavings,1,7);


        Button btnLoan= new Button("Loans");
        btnLoan.setFont(Font.font("Arial", FontWeight.BOLD,24));
        gridMainPage.add(btnLoan,1,9);


        Button btnMortgage= new Button("Mortgage");
        btnMortgage.setFont(Font.font("Arial", FontWeight.BOLD,24));
        gridMainPage.add(btnMortgage,1,11);


        Button btnHistory= new Button("History");
        gridMainPage.add(btnHistory,1,13);
        btnHistory.setFont(Font.font("Arial", FontWeight.BOLD,24));




    }
}


Financial Calculator/src/sample/LoansHelpView.java
package sample;

import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.VBox;
import javafx.scene.text.Font;
import javafx.scene.text.FontWeight;
import javafx.stage.Modality;
import javafx.stage.Stage;

public class LoansHelpView {


    public static void displayLoanHelp()
    {
        Stage popupwindowLoan=new Stage();

        popupwindowLoan.initModality(Modality.APPLICATION_MODAL);
        popupwindowLoan.setTitle("Loan Help View");


        Label labelLoan1= new Label("\tHow to use Loan Calculator ");
        labelLoan1.setFont(Font.font("Verdana", FontWeight.BOLD, 24));
        Label labelLoan2= new Label("\tCalculate Monthly Payment\n");
        labelLoan2.setFont(Font.font(20));
        Label labelLoan3= new Label("\tTo calculate Monthly of your Loan please input \n" +
                "\t\t* The loan amount in rupees \n" +
                "\t\t* invested time period in years as Loan Term\n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* If there is a down payment, input the value to the appropriate text box\n"+
                "\t\t* and leave space on Monthly Payment input.\n" +
                "\tAfter entering appropriate inputs then choose Monthly Payment from Solve for drop down list and click on Calculate button.\n");

        Label labelLoan4= new Label("\tLoan Period\n");
        labelLoan4.setFont(Font.font(20));
        Label labelLoan5= new Label("\tTo calculate Yearly Interest rate  of your loan please input \n" +
                "\t\t* The loan amount in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* If there is a down payment, input the value to the appropriate text box\n"+
                "\t\t* Input the Monthly Payment in rupees.\n" +
                "\t\t* and leave space on Loan Period input.\n" +
                "\tAfter entering appropriate inputs then choose Loan Period from Solve for drop down list and click on Calculate button.\n ");

        Label labelLoan6= new Label("\tPresent Value \n");
        labelLoan6.setFont(Font.font(20));
        Label labelLoan7= new Label("\tTo calculate the Invested time of your loan in years please input \n" +
                "\t\t* Input the Monthly Payment in rupees.\n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* Input the Monthly Payment in rupees.\n" +
                "\t\t* If there is a down payment, input the value to the appropriate text box\n"+
                "\t\t* and leave space on Loan Amount input.\n" +
                "\tAfter entering appropriate inputs then choose Present Value from Solve for drop down list and click on Calculate button. \n");




        Button buttonLoan1= new Button("Back");
        //button1.setLayoutX(1120);
        //button1.setLayoutY(30);


        buttonLoan1.setOnAction(e -> popupwindowLoan.close());



        VBox layoutLoan= new VBox(10);


        layoutLoan.getChildren().addAll(labelLoan1,labelLoan2,labelLoan3,labelLoan4,labelLoan5,labelLoan6,labelLoan7,buttonLoan1);

        layoutLoan.setAlignment(Pos.TOP_LEFT);

        Scene scene1= new Scene(layoutLoan, 900, 700);

        popupwindowLoan.setScene(scene1);

        popupwindowLoan.showAndWait();

    }

}




Financial Calculator/src/sample/MortgageHelpView.java
package sample;

import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.VBox;
import javafx.scene.text.Font;
import javafx.scene.text.FontWeight;
import javafx.stage.Modality;
import javafx.stage.Stage;

public class MortgageHelpView {


    public static void displayMortgageHelp()
    {
        Stage popupwindowMort=new Stage();

        popupwindowMort.initModality(Modality.APPLICATION_MODAL);
        popupwindowMort.setTitle("Mortgage Help View");


        Label labelMort1= new Label("\tHow to use Mortgage Calculator\n ");
        labelMort1.setFont(Font.font("Verdana", FontWeight.BOLD, 24));
        Label labelMort2= new Label("\tTime Period");
        labelMort2.setFont(Font.font(20));
        Label labelMort3= new Label("\tTo calculate Time Period of your mortgage please input \n" +
                "\t\t* The Home Price of your mortgage in rupees \n" +
                "\t\t* If there is a down payment input the value on Down Payment input area or put zero on the down payment input.\n"+
                "\t\t* The monthly payment of your mortgage in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\tAfter entering appropriate inputs then choose Time Period from Solve for drop down list and click on Calculate button.\n" +
                "\tThen you can see the results of your calculation in the right side of the calculator \n");

        Label labelMort4= new Label("\tMonthly Payment ");
        labelMort4.setFont(Font.font(20));
        Label labelMort5= new Label("\tTo calculate Your monthly mortgage payment please input \n" +
                "\t\t* The Home Price of your mortgage in rupees \n" +
                "\t\t* If there is a down payment input the value on Down Payment input area or put zero on the down payment input.\n"+
                "\t\t* invested time period in years\n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* and leave space on Monthly Payment input.\n" +
                "\tAfter entering appropriate inputs then choose Monthly Payment from Solve for drop down list and click on Calculate button.\n ");


        Button buttonMort1= new Button("Back");

        //button1.setLayoutX(1120);
        //button1.setLayoutY(30);


        buttonMort1.setOnAction(e -> popupwindowMort.close());



        VBox layout= new VBox(10);


        layout.getChildren().addAll(labelMort1,labelMort2,labelMort3,labelMort4,labelMort5,buttonMort1);

        layout.setAlignment(Pos.TOP_LEFT);

        Scene sceneMortHelp= new Scene(layout, 900, 750);

        popupwindowMort.setScene(sceneMortHelp);

        popupwindowMort.showAndWait();

    }

}



Financial Calculator/src/sample/SimpleSavingHelpView.java
package sample;

import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.VBox;
import javafx.scene.text.Font;
import javafx.scene.text.FontWeight;
import javafx.stage.Modality;
import javafx.stage.Stage;

public class SimpleSavingHelpView {


    public static void displaySavingHelp()
    {
        Stage popupwindowSav=new Stage();

        popupwindowSav.initModality(Modality.APPLICATION_MODAL);
        popupwindowSav.setTitle("\nSimple Saving Help View");


        Label labelSav1= new Label("\tHow to use Simple Saving Calculator ");
        labelSav1.setFont(Font.font("Verdana", FontWeight.BOLD, 24));
        Label labelSav2= new Label("\tCalculate Future Value ");
        labelSav1.setFont(Font.font(20));
        Label labelSav3= new Label("\tTo calculate future value of your saving please input \n" +
                "\t\t* The amount of money(Principal value) that your going to invest or already invested in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* invested time period in years\n" +
                "\t\t* and leave space on Future value input.\n" +
                "\tAfter entering appropriate inputs then choose Future Value from Solve for drop down list and click on Calculate button. ");

        Label labelSav4= new Label("\tCalculate Interest Rate ");
        labelSav4.setFont(Font.font(20));
        Label labelSav5= new Label("\tTo calculate Yearly Interest rate  of your saving please input \n" +
                "\t\t* The amount of money(Principal value) that your going to invest or already invested in rupees \n" +
                "\t\t* The future value of your saving in rupees \n" +
                "\t\t* invested time period in years\n" +
                "\t\t* and leave space on Interest Rate input.\n" +
                "\tAfter entering appropriate inputs then choose Interest Rate from Solve for drop down list and click on Calculate button. ");

        Label labelSav6= new Label("\tCalculate Invested Time ");
        labelSav6.setFont(Font.font(20));
        Label labelSav7= new Label("\tTo calculate the Invested time of your savings in years please input \n" +
                "\t\t* The amount of money(Principal value) that your going to invest or already invested in rupees \n" +
                "\t\t* The future value of your saving in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* The amount of money(Principal value) that your going to invest or already invested in rupees \n" +
                "\t\t* and leave space on Time Period input.\n" +
                "\tAfter entering appropriate inputs then choose Time Period from Solve for drop down list and click on Calculate button. ");

        Label labelSav8= new Label("\tCalculate the Invested money as a Simple Saving ");
        labelSav8.setFont(Font.font(20));
        Label labelSav9= new Label("\tTo calculate the Invested money  of your savings in years please input \n" +
                "\t\t* The future value of your saving in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* invested time period in years\n" +
                "\t\t* and leave space on Start Principal input.\n" +
                "\tAfter entering appropriate inputs then choose Start Principal from Solve for drop down list and click on Calculate button. ");


        Button buttonSav1= new Button("Back");
        //button1.setLayoutX(1120);
        //button1.setLayoutY(30);


        buttonSav1.setOnAction(e -> popupwindowSav.close());



        VBox layoutSav= new VBox(10);


        layoutSav.getChildren().addAll(labelSav1 ,labelSav2,labelSav3,labelSav4,labelSav5,labelSav6,labelSav7,labelSav8,labelSav9,buttonSav1);

        layoutSav.setAlignment(Pos.TOP_LEFT);

        Scene sceneSavHelp= new Scene(layoutSav, 900, 900);

        popupwindowSav.setScene(sceneSavHelp);

        popupwindowSav.showAndWait();

    }

}




Financial Calculator/src/sample/CompoundSavingHelpView.java
package sample;

import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.VBox;
import javafx.scene.text.Font;
import javafx.scene.text.FontWeight;
import javafx.stage.Modality;
import javafx.stage.Stage;

public class CompoundSavingHelpView {


    public static void displayCompHelp()
    {
        Stage popupwindowComp=new Stage();

        popupwindowComp.initModality(Modality.APPLICATION_MODAL);
        popupwindowComp.setTitle("Compound Saving Help View");


        Label labelComp1= new Label("\tHow to use Compound Saving Calculator ");
        labelComp1.setFont(Font.font("Verdana", FontWeight.BOLD, 24));
        Label labelComp2= new Label("\tCalculate Future Value ");
        labelComp2.setFont(Font.font(20));
        Label labelComp3= new Label("\tTo calculate future value of your compound saving please input \n" +
                "\t\t* The amount of money(Principal Investment) that your going to invest or already invested in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* invested time period in years\n" +
                "\t\t* Monthly payment in rupees\n" +
                "\t\t* and leave space on Future value input.\n" +
                "\tAfter entering appropriate inputs then choose Future Value from Solve for drop down list and click on Calculate button. ");

        Label labelComp4= new Label("\tCalculate Monthly Payment ");
        labelComp4.setFont(Font.font(20));
        Label labelComp5= new Label("\tTo calculate  the monthly payment of your compound saving please input \n" +
                "\t\t* The amount of money(Principal Investment) that your going to invest or already invested in rupees \n" +
                "\t\t* The future value of your saving in rupees \n" +
                "\t\t* invested time period in years\n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* and leave space on Monthly Payment input.\n" +
                "\tAfter entering appropriate inputs then choose Monthly Payment from Solve for drop down list and click on Calculate button. ");

        Label labelComp6= new Label("\tCalculate Present Value ");
        labelComp6.setFont(Font.font(20));
        Label labelComp7= new Label("\tTo calculate the Present Value of your Compound savings please input \n" +
                "\t\t* The future value of your saving in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* Monthly payment in rupees\n" +
                "\t\t* invested time period in years\n" +
                "\t\t* and leave space on Principal investment input.\n" +
                "\tAfter entering appropriate inputs then choose Present Value from Solve for drop down list and click on Calculate button. ");

        Label labelComp8= new Label("\tCalculate the Invested Time Period ");
        labelComp8.setFont(Font.font(20));
        Label labelComp9= new Label("\tTo calculate the Invested  Time Period of your compound savings in years please input \n" +
                "\t\t* The future value of your saving in rupees \n" +
                "\t\t* Annual interest rate as a percentage\n" +
                "\t\t* Monthly payment in rupees\n" +
                "\t\t* The amount of money(Principal Investment) that your going to invest or already invested in rupees \n" +
                "\t\t* and leave space on Time Period input.\n" +
                "\tAfter entering appropriate inputs then choose Time period from Solve for drop down list and click on Calculate button. ");


        Button buttonComp1= new Button("Back");
        //button1.setLayoutX(1120);
        //button1.setLayoutY(30);


        buttonComp1.setOnAction(e -> popupwindowComp.close());


        VBox layout= new VBox(10);


        layout.getChildren().addAll(labelComp1,labelComp2,labelComp3,labelComp4,labelComp5,labelComp6,labelComp7,labelComp8,labelComp9,buttonComp1);

        layout.setAlignment(Pos.TOP_LEFT);

        Scene CompHelp= new Scene(layout, 900, 900);

        popupwindowComp.setScene(CompHelp);

        popupwindowComp.showAndWait();

    }

}

